src/common.c

```diff
@@ -264,6 +264,17 @@ const char *ofi_straddr(char *buf, size_t *len,
 	case FI_ADDR_MLX:
 		size = snprintf(buf, *len, "fi_addr_mlx://%p", addr);
 		break;
+	case FI_ADDR_IB_UD:
+		size = snprintf(buf, *len, "fi_addr_ib_ud://"
+				"%" PRIx16 /* LID */ ":%" PRIx8 /* SL */
+				":%" PRIx64 /* GID::GUID */ ":%" PRIx64 /* GID::EUI */
+				":%" PRIx32 /* QPN */ ":%" PRIx32 /* PSN */ ":%" PRIx16 /* P_Key */
+				":%" PRIx16 /* NS service */,
+				*(uint16_t *)addr + 12, *(uint8_t *)addr + 30,
+				*(uint64_t *)addr, *(uint64_t *)addr + 1,
+				*(uint32_t *)addr + 4, *(uint32_t *)addr + 5, *(uint16_t *)addr + 13,
+				*(uint16_t *)addr + 14);
```

Comment:
I'm having a hard time figuring out what the "proper" format for this should be.  I think it makes sense for a gid to match up with ipv6, however.  I don't think psn is needed, so I would drop it.  The 16-bit NS isn't clear to me either.  That's not part of the IB address, so I would drop that as well.  (We have IPv4 and IPv6 formats.)  So... I *think* this format would be closer:
```
fi_addr_ib_ud://[gid]%qpn/pkey/lid/sl
```
Unfortunately I can't find a good example of extending an IP address with QoS information.  Everything beyond the ip%port ends up at a file system path.  The order of pkey, lid, and sl I listed is mostly arbitrary.

@jsquyres - Do you have any thoughts on how to include other parts of the address in this?